python/pycylon/pycylon/data/table.pyx

```diff
@@ -1197,6 +1206,29 @@ cdef class Table:
             final_table = Table.from_list(self.context, col_names, filtered_all_data)
             return final_table
 
+    cdef _get_slice_ra_response(self, offset, length, ra_op_name):
+        cdef:
+            shared_ptr[CTable] output
+            CStatus status
+
+        if ra_op_name == DISTRIBUTED_SLICE:
+            status = DistributedSlice(self.table_shd_ptr, offset, length, &output)
+        elif ra_op_name == DISTRIBUTED_HEAD:
+            status = DistributedHead(self.table_shd_ptr, length, &output)
+        elif ra_op_name == DISTRIBUTED_TAIL:
+            status = DistributedTail(self.table_shd_ptr, length, &output)
+        elif ra_op_name == HEAD:
+            status = Head(self.table_shd_ptr, length, &output)
+        elif ra_op_name == TAIL:
+            status = Tail(self.table_shd_ptr, length, &output)
+        else:
+            raise ValueError(f"Unsupported relational algebra operator: {ra_op_name}")
+
+        if status.is_ok():
+            return pycylon_wrap_table(output)
+        else:
+            raise ValueError(f"Head operation failed : {status.get_msg().decode()}")
+
```

Comment:
I don't think you understood my previous comment. I explicitly showed how to call a `cdef` function within another function. Why cannot we do that here. Either way I don't mind this approach if @nirandaperera is fine with this.

Reply:
@vibhatha , Thank you for the suggestion. You are right about the cython example. I was trying differently. But In the last meeting, I and @nirandaperera , discussed this approach and I implemented it based on that. I believe it's not a big deal.

Reply:
Sounds good to me.